### ISSUE TITLE
Support the unique ID docker 1.12 passes to match mount/unmount requests

### ISSUE THREAD

**Issue by govint** [vmware/docker-volume-vsphere#539]
New docker feature to pass a unique ID on mount and unmount reqs to ensure that mount/unmount requests are matched.

**Comment by govint** [comment]
Tried out printing the ID thats passed on mount/unmount , getting a null string from docker. Debugging the plugin-helper and docker (its 1.12 rc2). Will try a newer 1.12 RC.

2016-07-25 22:55:09.59181396 -0700 PDT [INFO] Mounting volume name="dvol_2"
2016-07-25 22:55:09.591921092 -0700 PDT [INFO] Mount - name="dvol_2" mount-id=

**Comment by govint** [comment]
Depends on #542

**Comment by govint** [comment]
Made an issue on go-plugin-helpers to fix the request structure to allow mount IDs to get copied correctly - issue 55

**Comment by govint** [comment]
Made an issue to docker (25331), the opaque ID doesn't seem to be coming in from docker on the unmount.

**Comment by govint** [comment]
Issues in go-plugin-helpers are fixed and raised separate PR to update into repo.

**Comment by govint** [comment]
Docker issue 25331 is fixed and changes should be available in the next RC. As discussed this issue isn't really fixing a bug thats not already addressed by volume ref counting in the plugin. Lowering priority on this issue and will take up later once we have a newer docker RC that has the fix for 25331.

**Comment by govint** [comment]
Closing this issue as these IDs are of no use in the plugin, The plugin implements ref counting itself which is working fine as is. Keeping multiple IDs per volume and matching mount to unmount seems way more costly than a single ref count per volume.
